src/event_handler.py

```diff
@@ -18,33 +18,46 @@ def store_queued_events(_, __):
     sqs_client = boto3.client('sqs')
     queue_url = os.environ['QUEUE_URL']
 
+    logger = logging.getLogger('event-recorder')
     if 'ENCRYPTION_KEY' in os.environ:
       encrypted_decryption_key = os.environ['ENCRYPTION_KEY']
+      logger.info('Got decryption key from envrionment variable')
     else:
       encrypted_decryption_key = fetch_decryption_key()
+      logger.info('Got decryption key from S3')
     decryption_key = decrypt(encrypted_decryption_key)
+    logger.info('Decrypted key successfully')
 
     database_password = None
     if 'ENCRYPTED_DATABASE_PASSWORD' in os.environ:
       # boto returns decrypted as b'bytes' so decode to convert to password string
       database_password = decrypt(os.environ['ENCRYPTED_DATABASE_PASSWORD']).decode()
     db_connection = create_db_connection(database_password)
+    logger.info('Created connection to DB')
 
+    event_count = 0
     while True:
+        event_count += 1
         message = fetch_single_message(sqs_client, queue_url)
         if message is None:
+            logger.info('Queue is empty - finishing after {0} events'.format(event_count))
             break
 
         # noinspection PyBroadException
         # catch all errors and log them - we never want a single failing message to kill the process.
         try:
             decrypted_message = decrypt_message(message['Body'], decryption_key)
             event = event_from_json(decrypted_message)
+            logger.info('Decrypted event with ID: {0}'.format(event.event_id))
             write_audit_event_to_database(event, db_connection)
+            logger.info('Stored audit event: {0}'.format(event.event_id))
             if event.event_type == 'session_event' and event.details.get('session_event_type') == 'idp_authn_succeeded':
+                logger.info('Stored billing event: {0}'.format(event.event_id))
```

Comment:
Should this come after the call to the DB?

Reply:
Yes ... fixed!